src/main/java/Duke.java

```diff
@@ -1,10 +1,117 @@
+    public static Task[] tasks;
+    public static Scanner in;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        initDuke();
+        greetUser();
+        while (isInUse) {
+            String userInput = getUserInput(in);
+            String[] processedInputs = processUserInput(userInput);
+            showResultToUser(tasks, processedInputs);
+        }
+    }
+
+    public static void initDuke() {
+        tasks = new Task[MAX_NUMBER_OF_TASKS];
+        in = new Scanner(System.in);
+    }
+
+    public static void showResultToUser(Task[] tasks, String[] processedInputs) {
+        if (processedInputs[0].equals("bye")) {
+            isInUse = false;
+            printExitMessage();
+        } else if (processedInputs[0].equals("list")) {
+            listTasks(tasks);
+        } else if (processedInputs[0].equals("mark")) {
+            int indexToBeMarked = Integer.parseInt(processedInputs[1]) - 1; // 0-indexed
+            tasks[indexToBeMarked].setDone(true);
+            printNotification(tasks[indexToBeMarked], "mark", tasksIndex + 1);
+        } else if (processedInputs[0].equals("unmark")) {
+            int indexToBeUnmarked = Integer.parseInt(processedInputs[1]) - 1; // 0-indexed
+            tasks[indexToBeUnmarked].setDone(false);
+            printNotification(tasks[indexToBeUnmarked], "unmark", tasksIndex + 1);
+        } else if (processedInputs[0].equals("todo")) {
+            tasks[tasksIndex] = new ToDo(processedInputs[1]);
+            printNotification(tasks[tasksIndex], "todo", tasksIndex + 1);
+            tasksIndex++;
+        } else if (processedInputs[0].equals("deadline")) { // now, if the array has 2 elements, then it is "deadline" case
+            String[] taskNameAndBy = furtherProcessInputForDeadline(processedInputs);
+            tasks[tasksIndex] = new Deadline(taskNameAndBy[0], taskNameAndBy[1]);
+            printNotification(tasks[tasksIndex], "deadline", tasksIndex + 1);
+            tasksIndex++;
+        } else { // must be "event" case
+            String[] fromAndToAndTaskName = furtherProcessInputForEvent(processedInputs);
+            tasks[tasksIndex] = new Event(fromAndToAndTaskName[2], fromAndToAndTaskName[0], fromAndToAndTaskName[1]);
+            printNotification(tasks[tasksIndex], "event", tasksIndex + 1);
+            tasksIndex++;
+        }
+    }
+
+    private static String[] furtherProcessInputForDeadline(String[] processedInputs) {
+        String[] taskNameAndIdentifierAndBy = processedInputs[1].split("/", 2);
+        String[] identifierAndBy = taskNameAndIdentifierAndBy[1].split(" ", 2);
+        String[] taskNameAndBy = {taskNameAndIdentifierAndBy[0], identifierAndBy[1]};
```

Comment:
Naming is very confusing should just name individual String variables like taskName, identifier and by.